"Revert "Adding documentation for transformer support for Image Classification"" (#246)
ml5js/ml5-website-v02-docsify: docs/reference/image-classifier.md

## Description

The ml5.js imageClassifier loads a pre-trained model that attempts to classify the content of an image. Each model can only predict labels from a fixed set of categories it was trained to recognize (typically objects, animals, and other visual categories). The image classifier uses a neural network to analyze the image and provide a list of possible labels for the content of the image in its entirety.

> The ml5.js imageClassifier uses MobileNet by default, but supports additional model architectures including transformer-based models and custom-trained models such as DoodleNet.

Reviewer: ```suggestion
The ml5.js imageClassifier uses MobileNet by default, but supports additional model architectures including transformer-based models, fine-tuned models such as `Food101`, specialized pre-trained models such as `DoodleNet`, and custom models created with [Teachable Machine](/reference/image-classifier-tm). MobileNet was trained on the 1,000 category ImageNet dataset.```

It provides the following functionality:

- **Image Classification**: ImageClassifier classifies the conents of a static image and provides a list of possible labels.

> - **Video Object Detection**: ImageClassifier continously classifies the conents of a static image and provides a list of possible labels.

Reviewer: ```suggestion
- ***Single Image Classification*** (`classify()`): Classifies one image and provides a list of possible labels.
- ***Continuous Video Classification*** (`classifyStart()`): Repeatedly classifies frames from a video and provides updated labels.
```

## Supported Models

ImageClassifier supports different neural network architectures depending on the model selected.

> - **MobileNet** (default): This model loads MobileNet, a lightweight convolutional neural network designed for efficient image classification. It was pretrained on ImageNet, a dataset of approximately 15 million images across 1,000 classes. MobileNet is optimized for relatively fast inference and is commonly used for real-time image classification in browser and mobile applications.

Reviewer: ```suggestion
- **MobileNet** (default): This model loads [MobileNet](https://github.com/tensorflow/models/tree/master/research/slim/nets/mobilenet), a lightweight convolutional neural network designed for efficient image classification. It was pretrained on ImageNet, a dataset of approximately 15 million images across 1,000 classes. MobileNet is optimized for relatively fast inference and is commonly used for real-time image classification in browser and mobile applications.
```

- **ViTBase**: This model loads Xenova/vit-base-patch16-224, which is a transformers.js version of Google’s vit-base-patch16-224 model. It was pretrained on ImageNet-21K: about 14 million images and 21,843 classes and then fine-tuned on ImageNet-1K to predict the familiar 1,000 classes. The Vision Transformer architecture divides an image into patches and uses an attention mechanism to learn relationships between different parts of the image.[Reference](https://huggingface.co/Xenova/vit-base-patch16-224)

> - **SwinFood101**: This model loads onnx-community/swin-finetuned-food101-ONNX, an ONNX version of a Swin Transformer fine-tuned on the Food101 dataset. It was fine-tuned from Microsoft’s swin-base-patch4-window7-224 model to classify images into 101 different food categories. Swin uses shifted windows of attention to efficiently learn relationships between different regions of an image. [Reference](https://huggingface.co/onnx-community/swin-finetuned-food101-ONNX)

Reviewer: ```suggestion
- **ViTBase**: This model loads [Xenova/vit-base-patch16-224](https://huggingface.co/Xenova/vit-base-patch16-224), which is a transformers.js version of Google’s vit-base-patch16-224 model. It was pretrained on [ImageNet-21K](https://www.image-net.org/): about 14 million images and 21,843 classes and then fine-tuned on ImageNet-1K to predict the familiar 1,000 classes. The Vision Transformer architecture divides an image into patches and uses an attention mechanism to learn relationships between different parts of the image.
- **SwinFood101**: This model loads [onnx-community/swin-finetuned-food101-ONNX](https://huggingface.co/onnx-community/swin-finetuned-food101-ONNX), an ONNX version of a Swin Transformer fine-tuned on the [Food101 dataset](https://huggingface.co/datasets/ethz/food101). It was fine-tuned from Microsoft’s swin-base-patch4-window7-224 model to classify images into 101 different food categories. Swin uses shifted windows of attention to efficiently learn relationships between different regions of an image.
```

I think it's best to link the models and datasets from within the paragraph. This is best for accessibility / screenreader users where generic "Reference" links don't provide context of what they are linking to.

> - **DoodleNet**; DoodleNet has its own set of 345 drawing categories from the Google Quick, Draw! dataset. (link to Yining's repo for training this model / Quick, Draw)

Reviewer: ```suggestion
- **DoodleNet**: [DoodleNet](https://github.com/yining1023/doodleNet) has its own set of 345 drawing categories from the [Google Quick, Draw! dataset](https://github.com/googlecreativelab/quickdraw-dataset).
```

- **Custom models**: Users can load compatible custom-trained models for specialized classification tasks.

> ?> Even though they use different neural network architectures, MobileNet and ViTBase predict from the same 1,000 ImageNet categories.

Reviewer: ```suggestion
Even though they use different neural network architectures, MobileNet and ViTBase predict from the same 1,000 ImageNet categories.
```

> ?> If you want to **train your own image classification model with customized labels**, check out our [Image + Teachable Machine](/reference/image-classifier-tm) to get started!

Reviewer: ```suggestion
If you want to **train your own image classification model with customized labels**, check out our [Image + Teachable Machine](/reference/image-classifier-tm) to get started!
```

> The ml5.js imageClassifier is a pre-trained model that tries to recognize the content of an image. It tries to guess the identities of objects, animals, and even people in a picture. The image classifier uses a neural network to analyze the image and provide a list of possible labels for the content of the image in its entirety.

Reviewer: ```suggestion
The ml5.js imageClassifier loads a pre-trained model that attempts to classify the content of an image. Each model can only predict labels from a fixed set of categories it was trained to recognize (typically objects, animals, and other visual categories). The image classifier uses a neural network to analyze the image and provide a list of possible labels for the content of the image in its entirety.
```

The ml5.js imageClassifier uses MobileNet by default, but supports additional model architectures including transformer-based models and custom-trained models such as DoodleNet.

It provides the following functionality:

- **Image Classification**: ImageClassifier tries to recognize the content of an image and provide a list of possible labels.

> - **Video Object Detection**: ImageClassifier can also be used to classify objects in a video stream.

Reviewer: I'm not sure this is needed, but I would at a minimum, rewrite this to be more clear. For example, the two pieces of functionality are really the `classify()` and `classifyStart()` functions. Both do the same thing, classify the contents on a single image, but `classifyStart()` initiates a process to do it over and over again on video frames. There is no "object detection" or "classifying objects" in this model, that is a different ML task.

Author: Let me know if you like the changes better

## Supported Models

> ImageClassifier can use different neural network architectures depending on the model selected.

Reviewer: ```suggestion
ImageClassifier supports different neural network architectures depending on the model selected.
```

- **MobileNet** (default): A lightweight convolutional neural network (CNN) designed for fast inference in the browser.
- **VisionTransformer**: A transformer-based image classification model that uses attention mechanisms to analyze relationships between different regions of an image. Vision Transformers can provide different performance and accuracy tradeoffs compared to CNN-based architectures.
- **Darknet**: Additional pre-trained CNN models available for image classification.

> - **Custom models**: Users can load compatible custom-trained models for specialized classification tasks.

Reviewer: This is waffling between a list of architectures and specific models, I think it's more clear to be a list of the specific models we support. 

* MobileNet
* ViTBase or VisionTransformer (whatever we choose to call it)
* Food101
* DoodleNet

And it's a useful place to be more specific and detailed about the models. For example, information for the VisionTransformer we might say something like:

> This model loads `Xenova/vit-base-patch16-224`, which is a transformers.js version of Google’s `vit-base-patch16-224` model. It was pretrained on ImageNet-21K: about 14 million images and 21,843 classes and then fine-tuned on ImageNet-1K to predict the familiar 1,000 classes. The Vision Transformer architecture divides an image into patches and uses an attention mechanism to learn relationships between different parts of the image.(with links to Hugging Face or Google documentation of the model)

And maybe for DoodleNet:

> DoodleNet has its own set of 345 drawing categories from the Google Quick, Draw! dataset. (link to Yining's repo for training this model / Quick, Draw)

Maybe somewhere on the page we include a sentence like:

> Even though they use different neural network architectures, MobileNet and VisionTransformer predict from the same 1,000 ImageNet categories.

I would also include background and links for MobileNet and Food101.

Should we deprecate DarkNet? I've never used it and maybe it's an unnecessary addition.

Author: Let me know if you like the changes better

**Parameters:**

- **modelName**: Optional.
  - String: Name of the underlying model to use. Possible values include:
    - `mobilenet`
    - `darknet` (28 MB in size)

>     - `darknet-tiny` (4 MB)

Reviewer: remove support for? Add Food101, etc.

**Parameters:**

- **modelName**: Optional.
  - String: Name of the underlying model to use. Possible values include:
    - `mobilenet`
    - `doodlenet`
    - `ViTBase`

>     - `SwinFood101`

Reviewer: It would be good to be consistent with capitalization. I believe internally ml5.js will accept upper or lower case, etc. both `"MobileNet"` and `"mobilenet"` are valid. Let's check the implementation and then use camel case for all of these.
```suggestion
    - `MobileNet`
    - `DoodleNet`
    - `ViTBase`
    - `SwinFood101`
```

    - or a URL to a compatible model file.

- **options**: Optional.
  Object: An object to change the default configuration of the model.

>   - The default options for the default `mobilenet` model are

Reviewer: ```suggestion
  Object: An object to change the default configuration of the model.
  - The default options for the default `MobileNet` model are
```

    ```
    {
      alpha: 1.0,
      topk: 3
    }
    ```

    - _version_: The MobileNet version to use. Default is 2.
    - _alpha_: The width multiplier for the MobileNet. Default is 1.0.
    - _topk_: The number of labels to return. Default is 3.

>   - The default options for `VisionTransformer` model are

Reviewer: ```suggestion
  - The default options for the `ViTBase` model are
```
